### PR TITLE
Add getAdmin in action Creator,reducer,use axios #1808

### LINKED ISSUE #1808: Add getAdmin in action Creator, reducer, use axios for api
**Actual Behaviour**

Currently admin is being set only in local react state using jquery ajax method.

**Expected Behaviour**

Integrate redux for admin state, which would allow users to use admin globally. Also, use axios for api for admin

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1808 

Had to close https://github.com/fossasia/chat.susi.ai/pull/1836, it was not able to detect new changes

Changes:

1. Created getAdmin function to fetch using axios.
2. Added getAdmin to Login.js and removed from StaticAppBar, TopBar
3. Created action creator and reducer for getAdmin
4. Added mockStore for enzyme for StaticAppBar

Demo Link:  https://pr-1842-1-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: src/apis/index.js [hunk 1]
```diff
@@ -36,6 +36,11 @@ export function getChangePassword(payload) {
   });
 }
 
+export function getAdmin() {
+  const url = `${API_URL}/${AUTH_API_PREFIX}/showAdminService.json}`;
```
Comment (akshatnitd) [inline src/apis/index.js]: Remove `}` at the end of the url. It causes API to fail.

Did it work for you?

Also, the correct demo link is `https://pr-1842-fossasia-susi-web-chat.surge.sh/`. Kindly update it.

Comment (fragm3) [reply src/apis/index.js]: @akshatnitd made the changes

**Comment by fragm3** [comment]
@akshatnitd I am unable to do npm run build due to queryString error

**Comment by fragm3** [comment]
@akshatnitd made the changes, it will work now, please review

**Comment by fragm3** [comment]
@akshatnitd the link is: https://pr-1842-1-fossasia-susi-web-chat.surge.sh. It is working now. Please review

File: src/components/Auth/Login/Login.react.js [hunk 2]
```diff
@@ -145,6 +145,23 @@ class Login extends Component {
             loading: false,
           });
         });
+      actions
```
Comment (akshatnitd) [inline src/components/Auth/Login/Login.react.js]: This implementation is buggy. The actions are async are the `showAdminService` API returns false. Make a separate function in this component for checking Admin and make the function call only after successful login.

**Comment by AakashMallik** [comment]
@fragm3 To fix the login dialog, pass the requenOpenLogin function from the App.js down to the staticAppBar via props. That would fix it. But @prateek76 has already opened a PR to fix it. Now you decide what to do. your call.

**Comment by fragm3** [comment]
@AakashMallik let him solve it, I have updated my branch, it will work

**Pull request closed without merging**